Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1114,25 +1114,29 @@ private extension PurchasesOrchestrator {
         self.attribution.unsyncedAdServicesToken { adServicesToken in
             _ = Task<Void, Never> {
                 let transaction = await self.transactionFetcher.firstVerifiedTransaction
-                guard let transaction = transaction, let jwsRepresentation = transaction.jwsRepresentation  else {
-                    self.customerInfoManager.customerInfo(appUserID: currentAppUserID,
-                                                          fetchPolicy: .cachedOrFetched) { result in
-                        self.operationDispatcher.dispatchOnMainThread {
-                            completion?(result.mapError(\.asPurchasesError))
-                        }
+                let appTransactionJWS = await self.transactionFetcher.appTransactionJWS
+
+                guard let transaction = transaction, let jwsRepresentation = transaction.jwsRepresentation else {
+                    // No receipt is found, just post the AppTransaction.
+                    self.backend.post(receipt: .empty,
+                                      productData: nil,
+                                      transactionData: .init(appUserID: currentAppUserID,
+                                                             source: .init(isRestore: true,
+                                                                           initiationSource: .restore)),
```

Comment:
Should we propagate the values in the method parameters instead of hardcoding them? Values can be different depending on whether the developer called `restorePurchases` or `syncPurchases`.

Reply:
Absolutely! Will get that updated 👨‍🔧